lib/screens/qr_code_screen.dart

```diff
@@ -0,0 +1,105 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:qr_flutter/qr_flutter.dart';
+import 'package:superformula_qr_code/models/seed_model.dart';
+import 'package:superformula_qr_code/services/api_service.dart';
+import 'package:superformula_qr_code/shared/app_routes.dart';
+import 'package:http/http.dart' as http;
+
+class QrCodeScreen extends StatefulWidget {
+  const QrCodeScreen({Key? key}) : super(key: key);
+
+  @override
+  State<QrCodeScreen> createState() => _QrCodeScreenState();
+}
+
+class _QrCodeScreenState extends State<QrCodeScreen> {
+  // To avoid unecessary rebuilds, initialize api call & countdown stream outside of the build widget
+  final Future<SeedModel> _future = ApiService().getSeed(http.Client());
+
+  static const int _count = 15;
+  final Stream<int> _countdown = countdown(_count);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('QR Code'),
+        centerTitle: true,
+      ),
+      body: StreamBuilder<dynamic>(
+          stream: _countdown,
+          builder: (context, snapshot) {
+            switch (snapshot.connectionState) {
```

Comment:
**suggestion**: all good here but you could improve the structure here a bit. Try to wrap the contents of the `case` statements in their own stateless or stateful widget to make them more readable AND to be able to use `const` constructors